### PR TITLE
feat: adds isValidAddress

### PR DESCRIPTION
**Pull request by natenichols**
## High Level Overview of Change
Adds `isValidAddress`, which validates that an address is either a valid X-Address or classic address. 

### Context of Change
Feature was requested by @mDuo13.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Test Plan
Added unit tests

### CODE REVIEW

File: test/utils/validAddress.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { assert } from 'chai'
```
Comment (mvadari) [inline test/utils/validAddress.ts:1]: nit: this file should be called `isValidAddress`

**Review by mvadari — APPROVED** [review]

File: test/utils/isValidAddress.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { assert } from 'chai'
+
+import { isValidAddress } from '../../src'
```
Comment (JST5000) [inline test/utils/isValidAddress.ts:3]: For browser builds this should import from 'xrpl-local'

**Review by JST5000 — APPROVED** [review]
LGTM other than the import thing

File: test/utils/isValidAddress.ts (hunk 2 above)
Comment (mvadari) [reply test/utils/isValidAddress.ts:3]: This test isn't run on the browser - it's a unit test, not an integration test.

**Pull request merged**